src/dkg/board.rs

```diff
@@ -0,0 +1,22 @@
+/// # Board
+///
+/// A board is where DKG participants publish their data for the corresponding DKG
+/// phase.
+use super::primitives::{BundledJustification, BundledResponses, BundledShares};
+use crate::group::Curve;
+
+/// Trait which must be implemented for writing to the board. This trait assumes
+/// an authenticated channel.
+pub trait BoardPublisher<C: Curve> {
+    type Error;
+
+    /// Publishes the shares to the board
+    fn publish_shares(&mut self, shares: BundledShares<C>) -> Result<(), Self::Error>;
```

Comment:
Note: I won't be able to use this trait unfortunately as I need these methods to be `async`, which is a little unfortunate.

Reply:
Yeah I have that in mind, considering moving this to `async_trait`. If you have any code which would potentially consume this, please share:)

Reply:
On further thought, we could have a `BoardPublisher` and `AsyncBoardPublisher` trait for that functionality.

Reply:
@gakonst sent you an invite to a repo, where I am using this atm